testutil/events_builder.go

```diff
@@ -0,0 +1,187 @@
+// Create one using NewEventsBuilder, then add events using the methods like AddEvent or AddSendCoins.
+// Once everything has been added, use Build() to get the final list of events.
+type EventsBuilder struct {
+	Events sdk.Events
+	T      *testing.T
+}
+
+// NewEventsBuilder creates a new EventsBuilder that will use the provided T for error checking.
+// If a T is provided, errors will cause test failures, otherwise errors will cause panics.
+func NewEventsBuilder(t *testing.T) *EventsBuilder {
+	return &EventsBuilder{T: t}
+}
+
+// Build returns the list of Events built so far.
+func (b *EventsBuilder) Build() sdk.Events {
+	return b.Events
+}
+
+// AddEvent adds one or more sdk.Event entries to this builder.
+func (b *EventsBuilder) AddEvent(events ...sdk.Event) *EventsBuilder {
+	b.Events = append(b.Events, events...)
+	return b
+}
+
+// AddEvents adds all of the provided sdk.Events to this builder.
+func (b *EventsBuilder) AddEvents(events sdk.Events) *EventsBuilder {
+	b.Events = append(b.Events, events...)
+	return b
+}
+
+// AddTypedEvent converts each of the provided events into an sdk.Event then adds it to this builder.
+func (b *EventsBuilder) AddTypedEvent(tevs ...proto.Message) *EventsBuilder {
+	if b.T != nil {
+		b.T.Helper()
+	}
+	var err error
+	events := make(sdk.Events, len(tevs))
+	for i, tev := range tevs {
+		events[i], err = sdk.TypedEventToEvent(tev)
+		switch {
+		case b.T != nil:
+			require.NoError(b.T, err, "TypedEventToEvent(%#v)", tev)
+		case err != nil:
+			panic(err)
+		}
+	}
+	return b.AddEvents(events)
+}
+
+// AddSendCoins adds the events emitted during SendCoins.
+func (b *EventsBuilder) AddSendCoins(from, to sdk.AccAddress, amount sdk.Coins) *EventsBuilder {
+	return b.AddEvents(SendCoinsEvents(from, to, amount))
+}
+
+// AddSendCoinsStrs adds the events emitted during SendCoins, but takes in the values as strings.
+func (b *EventsBuilder) AddSendCoinsStrs(from, to, amount string) *EventsBuilder {
+	return b.AddEvents(SendCoinsEventsStrs(from, to, amount))
+}
+
+// SendCoinsEvents creates the events emitted during SendCoins.
+func SendCoinsEvents(from, to sdk.AccAddress, amount sdk.Coins) sdk.Events {
+	return SendCoinsEventsStrs(from.String(), to.String(), amount.String())
+}
+
+// SendCoinsEventsStrs creates the events emitted during SendCoins, but takes in the values as strings.
+func SendCoinsEventsStrs(from, to, amount string) sdk.Events {
+	return sdk.Events{
+		{Type: "coin_spent", Attributes: []abci.EventAttribute{
+			{Key: "spender", Value: from},
+			{Key: "amount", Value: amount},
+		}},
+		{Type: "coin_received", Attributes: []abci.EventAttribute{
+			{Key: "receiver", Value: to},
+			{Key: "amount", Value: amount},
+		}},
+		{Type: "transfer", Attributes: []abci.EventAttribute{
+			{Key: "recipient", Value: to},
+			{Key: "sender", Value: from},
+			{Key: "amount", Value: amount},
+		}},
+		{Type: "message", Attributes: []abci.EventAttribute{
+			{Key: "sender", Value: from},
+		}},
+	}
+}
```

Comment:
Your suggestion doesn't help anything except increasing the amount of code. Your `sendCoinEvents` function is completely worthless because it's exactly what `SendCoinsEventsStrs` is now.